src/Plugin/resource/DataProvider/RateLimitedDataProvider.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * @file
+ * Contains Drupal\restful\Plugin\resource\DataProvider\RateLimitedDataProvider.
+ */
+
+namespace Drupal\restful\Plugin\resource\DataProvider;
+
+use Drupal\restful\Exception\ServerConfigurationException;
+use Drupal\restful\Plugin\resource\Field\ResourceFieldInterface;
+use Drupal\restful\RateLimit\RateLimitManagerInterface;
+
+class RateLimitedDataProvider implements DataProviderInterface {
```

Comment:
The DataProvider name is confusing here, as infact the RateLimit isn't really a data provider - it just a middleware. (Is the cache resource also a middleware?) Maybe we  actually need :

single resource 
single data provider
multiple middlewares ?